core/cache/model.go

```diff
@@ -128,19 +131,26 @@ func (m *Model) Machine(machineId string) (*Machine, error) {
 
 // WatchMachines returns a PredicateStringsWatcher to notify about
 // added and removed machines in the model.  The initial event contains
-// a slice of the current machine ids.
-func (m *Model) WatchMachines() *PredicateStringsWatcher {
+// a slice of the current machine ids.  Containers are excluded.
+func (m *Model) WatchMachines() (*PredicateStringsWatcher, error) {
 	m.mu.Lock()
 
+	// Create a compiled regexp to match machines not containers.
```

Comment:
Fixing the required lxd profiles based on juju lxd machine vs juju lxd container, brought to light the problem of WatchMachines returning containers too.  Makes QA easier to fix now.